inc/saiobject.h

```diff
@@ -61,6 +61,27 @@ typedef struct _sai_object_key_t
 
 } sai_object_key_t;
 
+/**
+ * @brief Structure for attribute capabilities per operation
+ */
+typedef struct _sai_attr_capability_t
+{
+    /**
+     * @brief Create is implemented
+     */
+    bool create_implemented;
+
+    /**
+     * @brief Set is implemented
+     */
+    bool set_implemented;
+
+    /**
+     * @brief Get is implemented
+     */
+    bool get_implemented;
```

Comment:
why remove is missing ? Is create implemented implies that remove is implemented as well?

Reply:
As for remove operation, remove operation doesn't accept an attribute list, remove operation is without any attribute, so checking whether an attribute is implemented for remove is not relevant

As for the reason having this API - we won't to avoid having to call create/get/set and get an error, and then understand that the attribute isn't implemented. AFAIK, for example, Sonic now crashes on such error. And such error also gets the log dirty. So we want a clean way for the application to query capabilities

Reply:
@kcudnik , one challenge of using create api is that it will taint the asic sometimes with failed operation. For example, to implement a feature, there are two approaches. We can try to use the first approach, it may fail at certain stage, then we have to roll back all previous actions and try the 2nd approach. The roll back operations might be challenage. 

Instead if we have the query API, we can query multiple capabilities at once and decide which way to go.

Reply:
of course, but as we spoke some time ago that when create fails rollback should be easy, rollback could be challenge when we would set multiple attributes at once thatswhy we have only one attribute for set api. And query first is way to go good enough

Reply:
yes, but an approach may require creating multiple objects, and you failed for the last object creation. then you need to rollout back all previous created objects. Sometime, the object cannot be created because the resource constraint not the asic capability, in this case you need to check the error code and why it fails, and then decide how to react.